pydoc/developer/BENCHMARK.md

```diff
@@ -0,0 +1,470 @@
+<tr>
+    <td>--size</td>
+    <td>10**8</td>
+</tr>
+<tr><td colspan="2">Problem size: length of array to use for benchmarks.</td></tr>
+<tr>
+    <td>--trials</td>
+    <td>5</td>
+</tr>
+<tr><td colspan="2">Number of times to run each test before averaging results. For tests that run as many trials as possible in 
+a given time, will be treated as number of seconds to run for.</td></tr>
+<tr>
+    <td>--seed</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Value to initialize random number generator.</td></tr>
+<tr>
+    <td>--dtype</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Dtypes to run benchmarks against. Comma separated list (NO SPACES) allowing for multiple. Accepted values: 
+int64, uint64, bigint, float64, bool, str and mixed. Mixed is used to generate sets of multiple types.</td></tr>
+<tr>
+    <td>--numpy</td>
+    <td>False</td>
+</tr>
+<tr><td colspan="2">When set, runs numpy comparison benchmarks.</td></tr>
+<tr>
+    <td>--maxbits</td>
+    <td>-1</td>
+</tr>
+<tr><td colspan="2">Only applies to bigint testing. Maximum number of bits, so values > 2**max_bits will wraparound. -1 is 
+interpreted as no maximum.</td></tr>
+<tr>
+    <td>--alpha</td>
+    <td>1.0</td>
+</tr>
+<tr><td colspan="2">Scalar multiple</td></tr>
+<tr>
+    <td>--randomize</td>
+    <td>False</td>
+</tr>
+<tr><td colspan="2">Fill arrays with random values instead of ones</td></tr>
+<tr>
+    <td>--index_size</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Only used for Gather and Scatter Benchmarks. Length of index array (number of gathers to perform)</td></tr>
+<tr>
+    <td>--value_size</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Only used for Gather and Scatter Benchmarks. Length of array from which values are gathered</td></tr>
+<tr>
+    <td>--encoding</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Only applies to encoding benchmarks. Comma separated list (NO SPACES) allowing for multiple encoding to be 
+used. Accepted values: idna, ascii</td></tr>
+<tr>
+    <td>--io_only_write</td>
+    <td>False</td>
+</tr>
+<tr><td colspan="2">Only applies to IO benchmarks. Only write the files; files will not be removed</td></tr>
+<tr>
+    <td>--io_only_read</td>
+    <td>False</td>
+</tr>
+<tr><td colspan="2">Only applies to IO benchmarks. Only read the files; files will not be removed</td></tr>
+<tr>
+    <td>--io_only_delete</td>
+    <td>False</td>
+</tr>
+<tr><td colspan="2">Only applies to IO benchmarks. Only delete files created from writing with this benchmark</td></tr>
+<tr>
+    <td>--io_files_per_loc</td>
+    <td>1</td>
+</tr>
+<tr><td colspan="2">Only applies to IO benchmarks. Number of files to create per locale</td></tr>
+<tr>
+    <td>--io_compression</td>
+    <td></td>
+</tr>
+<tr><td colspan="2">Only applies to IO benchmarks. Compression types to run Parquet IO benchmarks against. Comma delimited list 
+(NO SPACES) allowing for multiple. Accepted values: none, snappy, gzip, brotli, zstd, and lz4</td></tr>

```

Comment:
Can we provide an example of this being used?

Reply:
Added